lib/ecto/changeset.ex

```diff
@@ -791,4 +791,75 @@ defmodule Ecto.Changeset do
 
   defp too_long(length, value) when length <= value, do: nil
   defp too_long(_length, value), do: {:too_long, value}
+
+  @doc """
+  Validates the properties of a number.
+
+  ## Options
+
+    * `:less_than`
+    * `:greater_than`
+    * `:less_than_or_equal_to`
+    * `:greater_than_or_equal_to`
+    * `:equal_to`
+
+  ### Integer-Only
+
+    * `:parity`
+
+  ## Examples
+
+      validate_number(changeset, :count, less_than: 3, parity: even)
+      validate_number(changeset, :pi, greater_than: 3, less_than: 4)
+      validate_number(changeset, :the_answer_to_life_the_universe_and_everything, equal_to: 42)
+
+  """
+  @spec validate_number(t, atom, Range.t | [Keyword.t]) :: t
+  def validate_number(changeset, field, opts) do
+    float_validators = %{
+      less_than:                {&</2, :must_be_less_than},
+      greater_than:             {&>/2, :must_be_greater_than},
+      less_than_or_equal_to:    {&<=/2, :must_be_less_than_or_equal_to},
+      greater_than_or_equal_to: {&>=/2, :must_be_greater_than_or_equal_to},
+      equal_to:                 {&==/2, :must_be_equal_to},
+    }
+    integer_validators = Map.merge(float_validators, %{
+      parity:                   {&check_parity/2, :must_have_parity},
+    })
```

Comment:
If we move these to module attributes we dont have to `Map.merge` every time we call the function.